app/code/Magento/Catalog/Model/ProductRenderList.php

```diff
@@ -112,7 +112,7 @@ public function getList(\Magento\Framework\Api\SearchCriteriaInterface $searchCr
 
         $searchResult = $this->searchResultFactory->create();
         $searchResult->setItems($items);
-        $searchResult->setTotalCount(count($items));
+        $searchResult->setTotalCount($productCollection->getSize());
```

Comment:
Why we cannot pass it into constructor?

Reply:
Sorry, not getting your point. There is an issue with a total count. before it returns resulted items count. Which may be limited if set per page limit. So I just pass the total count of the collection.